File: daily-coding/coding-bat/string_splosion.py
# Given a non-empty string like "Code" return a string like "CCoCodCode".

def string_splosion(str):
  myList = list(str)
  myCopy = []
  myString = ''

  for x in range(0,len(myList)):
    myCopy += myList[0:x+1]

  myString = myString.join(myCopy)

  return myString

File: daily-coding/coding-bat/java/stringSplosion.java
// Given a non-empty string like "Code" return a string like "CCoCodCode".

public String stringSplosion(String str) {
  String[] myArray = str.split("");
  String newStr = "";
  int length = myArray.length;

  for(int i = 0; i < length; i++){
    for(int j = 0; j < i + 1; j++){
      newStr = newStr + myArray[j];
    }
  }
  return newStr;
}
